lib/msf/core/post/memory_search.rb

```diff
@@ -0,0 +1,150 @@
+# In your module, you can then call off to `memory_search`. :tada:
+module Msf::Post::MemorySearch
+  require 'rex/post/meterpreter/extensions/stdapi/tlv'
+
+  def initialize_memory_search(info = {}, platforms: %w[])
+    update_info(
+      info,
+      'Name' => 'Memory Search',
+      'Description' => %q{
+         This module allows for searching the memory space of running processes for
+         potentially sensitive data such as passwords.
+       },
+      'License' => MSF_LICENSE,
+      'Author' => %w[sjanusz-r7],
+      'SessionTypes' => %w[meterpreter],
+      'Platform' => platforms,
+      'Arch' => [ARCH_X86, ARCH_X64],
+      'Compat' => {
+        'Meterpreter' => {
+          'Commands' => %w[
+             stdapi_sys_process_memory_search
+             stdapi_sys_process_get_processes
+          ]
+        }
+      },
+      'Notes' => {
+        'Stability' => [::Msf::CRASH_SAFE],
+        'Reliability' => [],
+        'SideEffects' => []
+      }
+    )
+  end
+
+  def process_names_glob
+    datastore['PROCESS_NAMES_GLOB']
+  end
+
+  def process_ids
+    datastore['PROCESS_IDS']
+  end
+
+  def regex
+    datastore['REGEX']
+  end
+
+  def min_match_len
+    datastore['MIN_MATCH_LEN']
+  end
+
+  def max_match_len
+    datastore['MAX_MATCH_LEN']
+  end
+
+  def replace_non_printable_bytes?
+    datastore['REPLACE_NON_PRINTABLE_BYTES']
+  end
+
+  def save_loot?
+    datastore['SAVE_LOOT']
+  end
+
+  def memory_search(pid, needle, min_search_len, match_len)
+    request = ::Rex::Post::Meterpreter::Packet.create_request(::Rex::Post::Meterpreter::Extensions::Stdapi::COMMAND_ID_STDAPI_SYS_PROCESS_MEMORY_SEARCH)
+    request.add_tlv(::Rex::Post::Meterpreter::Extensions::Stdapi::TLV_TYPE_PID, pid)
+    request.add_tlv(::Rex::Post::Meterpreter::Extensions::Stdapi::TLV_TYPE_MEMORY_SEARCH_NEEDLE, needle)
+    request.add_tlv(::Rex::Post::Meterpreter::Extensions::Stdapi::TLV_TYPE_MEMORY_SEARCH_MATCH_LEN, match_len)
+    request.add_tlv(::Rex::Post::Meterpreter::TLV_TYPE_UINT, min_search_len)
+    client.send_request(request)
+  end
+
+  def print_result(result: nil)
+    return unless result
+
+    process_info = "#{result[:process]['name']} (pid: #{result[:process]['pid']})"
+    unless result[:status] == :success
+      print_warning "Memory search request for #{process_info} failed. Reason: #{result[:response]}"
+      return
+    end
+
+    result_group_tlvs = result[:response].get_tlvs(::Rex::Post::Meterpreter::Extensions::Stdapi::TLV_TYPE_MEMORY_SEARCH_RESULTS)
+    if result_group_tlvs.empty?
+      print_status "No regular expression matches were found in memory for #{process_info}"
+      return
+    end
+
+    results_table = ::Rex::Text::Table.new(
+      'Header' => "Memory Matches for #{process_info}",
+      'Indent' => 1,
+      'Columns' => ['Match Address', 'Match Length', 'Match Buffer', 'Memory Region Start', 'Memory Region Size']
+    )
+    x64_architectures = [ ARCH_X64, ARCH_AARCH64 ].freeze
```

Comment:
in the info hash you only support `'Arch' => [ARCH_X86, ARCH_X64],` so I don't think it can ever be `ARCH_AARCH64`